pkg/operations/remove.go

```diff
@@ -78,26 +75,34 @@ func RemoveVMContainer(result *runtime.ContainerInspectResult) {
 }
 
 // StopVM stops or kills a VM
-func StopVM(vm *api.VM, kill, silent bool) error {
+func StopVM(vm *api.VM, kill, silent, running bool) error {
 	var err error
 	container := util.NewPrefixer().Prefix(vm.GetUID())
 	action := "stop"
 
 	// Remove VM networking
-	if err = removeNetworking(util.NewPrefixer().Prefix(vm.GetUID()), true); err != nil {
-		return err
+	if running {
+		fmt.Printf("VM %q is not running but trying to cleanup networking for stopped container\n", vm.GetUID())
 	}
 
-	// Stop or kill the VM container
-	if kill {
-		action = "kill"
-		err = providers.Runtime.KillContainer(container, signalSIGQUIT) // TODO: common constant for SIGQUIT
-	} else {
-		err = providers.Runtime.StopContainer(container, nil)
+	if err = removeNetworking(util.NewPrefixer().Prefix(vm.GetUID()), running); err != nil {
+		return err
 	}
 
-	if err != nil {
-		return fmt.Errorf("failed to %s container for %s %q: %v", action, vm.GetKind(), vm.GetUID(), err)
+	if running {
+		// Stop or kill the VM container
+		if kill {
+			action = "kill"
+			err = providers.Runtime.KillContainer(container, signalSIGQUIT) // TODO: common constant for SIGQUIT
+		} else {
+			err = providers.Runtime.StopContainer(container, nil)
+		}
+
+		if err != nil {
+			return fmt.Errorf("failed to %s container for %s %q: %v", action, vm.GetKind(), vm.GetUID(), err)
+		}
+	} else {
+		return nil
```

Comment:
We can simplify this whole conditional block by returning nil if not running, reducing the level of indentation and keeping the code as it was before.